ios/FluentUI/Vnext/Button/MSFButton.swift

```diff
@@ -8,91 +8,38 @@ import SwiftUI
 import UIKit
 
 /// UIKit wrapper that exposes the SwiftUI Button implementation
-@objc open class MSFButton: NSObject,
-                            FluentUIWindowProvider,
-                            UIGestureRecognizerDelegate {
+@objc public class MSFButton: ControlHostingContainer,
+                              UIGestureRecognizerDelegate {
 
     /// Closure that handles the button tap event.
-    @objc open var action: ((_ sender: MSFButton) -> Void)?
+    @objc public var action: ((_ sender: MSFButton) -> Void)?
 
     /// The object that groups properties that allow control over the button appearance.
-    @objc open var state: MSFButtonState {
-        return self.buttonView.state
-    }
-
-    /// The UIView representing the button.
-    @objc open var view: UIView {
-        return hostingController.view
-    }
-
-    /// Creates a new MSFButton instance.
-    /// - Parameters:
-    ///   - style: The MSFButtonStyle used by the button. Defaults to secondary.
-    ///   - size: The MSFButtonSize value used by the button. Defaults to large.
-    ///   - action: Closure that handles the button tap event.
-    @objc public convenience init(style: MSFButtonStyle = .secondary,
-                                  size: MSFButtonSize = .large,
-                                  action: ((_ sender: MSFButton) -> Void)?) {
-        self.init(style: style,
-                  size: size,
-                  theme: nil,
-                  action: action)
-    }
+    @objc public let state: MSFButtonState
```

Comment:
What's the reason on not making this a computed property?
Although the state object shouldn't be changed through the lifecycle of the control, it might be error prone to rely on updating this property if it is recreated.

Reply:
Great question! This is part of a separation in the relationship between `state` and `tokens`, since the latter (`tokens`) is no longer stored, but generated itself. It's not really clear how `state` could maintain a reference to that generation logic, so now it simply has no concept of design tokens.

Additionally, those values have now become read-only after initialization, taking a page from UIKit controls such as `UIButton`, which are initialized with an immutable style. This won't affect our SwiftUI control, which of course recreates itself in full whenever its `style` or `size` are detected to have changed.

Reply:
As we've been discussing offline, this seems problematic because the tokens need to vary if "style" and "size" enum properties on the state object change.
Looking forward for the upcoming PR to address that.

Reply:
PR is out now! 

- https://github.com/microsoft/fluentui-apple/pull/852